modules/contact/src/main/kotlin/edu/stanford/spezi/modules/contact/component/AddressCard.kt

```diff
@@ -40,24 +39,26 @@ fun NavigationCard(address: String, publisher: (OnAction) -> Unit) {
                 .height(IntrinsicSize.Min),
             verticalAlignment = Alignment.CenterVertically
         ) {
-            Icon(
-                Icons.Default.Place,
-                contentDescription = "Address",
-            )
+            val addressText = address.formatted()
             Text(
-                text = address,
+                text = addressText,
                 style = TextStyles.bodyMedium,
                 modifier = Modifier.weight(1f)
             )
+            val context = LocalContext.current
             IconButton(
                 onClick = {
-                    publisher(OnAction.NavigateTo(address = address))
+                    val addressQuery = URLEncoder.encode(addressText, "utf-8")
+                    val gmmIntentUri = Uri.parse("geo:0,0?q=$addressQuery")
+                    val mapIntent = Intent(Intent.ACTION_VIEW, gmmIntentUri)
+                    mapIntent.setPackage("com.google.android.apps.maps")
+                    context.startActivity(mapIntent)
```

Comment:
consider wrapping this inside of a `runCatching` as it can throw in case of maps app is not installed in the device

Reply:
Makes sense - I'd assume to at least log when an error occurs - can you guide me through the process of how to do this in this context? I know, I could just use `println`, but I'm assuming you use some other logging mechanism?

Reply:
I used `println` for now - let me know what you prefer here instead.

Reply:
yeah exactly, we already have a logging mechanism, see https://github.com/StanfordSpezi/SpeziKt/tree/main/core/logging, that module is added by default via the convention plugin as a dependency to every module, so feel free to use directly